packages-exp/auth-exp/src/core/strategies/credential.ts

```diff
@@ -39,3 +46,52 @@ export async function signInWithCredential(
   await auth.updateCurrentUser(userCredential.user);
   return userCredential;
 }
+
+export async function linkWithCredential(
+  userExtern: externs.User,
+  credentialExtern: externs.AuthCredential
+): Promise<UserCredential> {
+  const user = userExtern as User;
+  const credential = credentialExtern as AuthCredential;
+  await _assertLinkedStatus(false, user, credential.providerId);
+
+  const response = await credential._linkToIdToken(
+    user.auth,
+    await user.getIdToken()
+  );
+
+  const newCred = _authCredentialFromTokenResponse(response);
+  await user._updateTokensIfNecessary(response, /* reload */ true);
+  return new UserCredentialImpl(user, newCred, OperationType.LINK);
```

Comment:
shouldn't this be `newCred || credential`?

Reply:
No:
https://github.com/firebase/firebase-js-sdk/blob/712dfde546a89af6778ab2f6c787bfd47abbd3b4/packages/auth/src/authuser.js#L1330